### PR TITLE
[ja] Translate docs/concepts/storage/storage-classes/ into Japanese

### LINKED ISSUE #38147: [ja] Translate `docs/concepts/storage/storage-classes/` into Japanese
**This is a Feature Request**

**What would you like to be added**

Translate https://kubernetes.io/docs/concepts/storage/storage-classes/ into Japanese.
**Why is this needed**

**Comments**

/language ja
/assign

### PR DESCRIPTION
**Pull request by KobayashiD27**
close #38147

### CODE REVIEW

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+        - containerPort: 80
+          name: "http-server"
+      volumeMounts:
+        - mountPath: "/usr/share/nginx/html"
+          name: task-pv-storage
+```
+
+### 許可されたトポロジー
+
+クラスタオペレータが`WaitForFirstConsumer`ボリュームバインディングモードを指定すると、ほとんどの状況でプロビジョニングを特定のトポロジに制限する必要がなくなります。ただし、それでも必要な場合は、`allowedTopologies`を指定できます。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/storage-classes.md:162]:
```suggestion
クラスタオペレーターが`WaitForFirstConsumer`ボリュームバインディングモードを指定すると、ほとんどの状況でプロビジョニングを特定のトポロジに制限する必要がなくなります。ただし、それでも必要な場合は、`allowedTopologies`を指定できます。
```

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+- matchLabelExpressions:
+  - key: failure-domain.beta.kubernetes.io/zone
+    values:
+    - us-central-1a
+    - us-central-1b
+```
+
+## パラメーター
+
+ストレージクラスには、ストレージクラスに属するボリュームを記述するパラメーターがあります。`プロビジョナー`に応じて、異なるパラメーターが受け入れられる場合があります。たとえば、パラメータ`type`の値`io1`とパラメータ`iopsPerGB`はEBSに固有です。パラメータを省略すると、デフォルトが使用されます。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/storage-classes.md:185]:
```suggestion
ストレージクラスには、ストレージクラスに属するボリュームを記述するパラメーターがあります。`プロビジョナー`に応じて、異なるパラメーターが受け入れられる場合があります。たとえば、パラメーター`type`の値`io1`とパラメーター`iopsPerGB`はEBSに固有です。パラメーターを省略すると、デフォルトが使用されます。
```

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 3]
```diff
@@ -0,0 +1,538 @@
...
+
+StorageClassに定義できるパラメータは最大512個です。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/storage-classes.md:187]:
```suggestion
StorageClassに定義できるパラメーターは最大512個です。
```

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+  name: slow
+provisioner: kubernetes.io/aws-ebs
+parameters:
+  type: io1
+  iopsPerGB: "10"
+  fsType: ext4
+```
+
+* `type`:`io1`、`gp2`、`sc1`、`st1`。詳細については、[AWSドキュメント](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html)を参照してください。デフォルト:`gp2`。
+* `zone`(非推奨):AWS zone。`zone`も`zones`も指定されていない場合、ボリュームは通常、Kubernetesクラスターにノードがあるすべてのアクティブなゾーンにわたってラウンドロビン方式で処理されます。`zone`パラメータと`zones`パラメータを同時に使用することはできません。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/storage-classes.md:205]:
```suggestion
* `zone`(非推奨):AWS zone。`zone`も`zones`も指定されていない場合、ボリュームは通常、Kubernetesクラスターにノードがあるすべてのアクティブなゾーンにわたってラウンドロビン方式で処理されます。`zone`パラメーターと`zones`パラメーターを同時に使用することはできません。
```

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 5]
```diff
@@ -0,0 +1,538 @@
...
+* `zones`(非推奨):AWS zoneのコンマ区切りリスト。`zone`も`zones`も指定されていない場合、ボリュームは通常、Kubernetesクラスターにノードがあるすべてのアクティブなゾーンにわたってラウンドロビン方式で処理されます。`zone`パラメータと`zones`パラメータを同時に使用することはできません。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/storage-classes.md:206]:
```suggestion
* `zones`(非推奨):AWS zoneのコンマ区切りリスト。`zone`も`zones`も指定されていない場合、ボリュームは通常、Kubernetesクラスターにノードがあるすべてのアクティブなゾーンにわたってラウンドロビン方式で処理されます。`zone`パラメーターと`zones`パラメーターを同時に使用することはできません。
```

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 6]
```diff
@@ -0,0 +1,538 @@
...
+* `iopsPerGB`:`io1`ボリュームのみ。GiBごとの1秒あたりのI/O操作。AWSボリュームプラグインは、これを要求されたボリュームのサイズで乗算して、ボリュームのIOPSを計算し、上限を20,000IOPSに設定します(AWSでサポートされる最大値については、[AWSドキュメント](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html)を参照してください)。ここでは文字列が必要です。つまり、`10`ではなく`"10"`です。
+* `fsType`:kubernetesでサポートされているfsType。デフォルト:`"ext4"`。
+* `encrypted`:EBSボリュームを暗号化するかどうかを示します。有効な値は`"true"`または`"false"`です。ここでは文字列が必要です。つまり、`true`ではなく`"true"`です。
+* `kmsKeyId`:オプション。ボリュームを暗号化するときに使用するキーの完全なAmazonリソースネーム。何も指定されていなくても`encrypted`がtrueの場合、AWSによってキーが生成されます。有効なARN値については、AWSドキュメントを参照してください。
+
+{{< note >}}
+`zone`および`zones`パラメータは廃止され、[allowedTopologies](#allowed-topologies)に置き換えられました。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/storage-classes.md:213]:
```suggestion
`zone`および`zones`パラメーターは廃止され、[allowedTopologies](#allowed-topologies)に置き換えられました。
```

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 7]
```diff
@@ -0,0 +1,538 @@
...
+  name: slow
+provisioner: kubernetes.io/gce-pd
+parameters:
+  type: pd-standard
+  fstype: ext4
+  replication-type: none
+```
+
+* `type`:`pd-standard`または`pd-ssd`。デフォルト:`pd-standard`
+* `zone`(非推奨):GCE zone。`zone`も`zones`も指定されていない場合、ボリュームは通常、Kubernetesクラスターにノードがあるすべてのアクティブなゾーンにわたってラウンドロビン方式で処理されます。`zone`パラメータと`zones`パラメータを同時に使用することはできません。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/storage-classes.md:231]:
```suggestion
* `zone`(非推奨):GCE zone。`zone`も`zones`も指定されていない場合、ボリュームは通常、Kubernetesクラスターにノードがあるすべてのアクティブなゾーンにわたってラウンドロビン方式で処理されます。`zone`パラメーターと`zones`パラメーターを同時に使用することはできません。
```

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 8]
```diff
@@ -0,0 +1,538 @@
...
+* `zones`(非推奨):GCE zoneのコンマ区切りリスト。`zone`も`zones`も指定されていない場合、ボリュームは通常、Kubernetesクラスターにノードがあるすべてのアクティブなゾーンにわたってラウンドロビン方式で処理されます。`zone`パラメータと`zones`パラメータを同時に使用することはできません。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/storage-classes.md:232]:
```suggestion
* `zones`(非推奨):GCE zoneのコンマ区切りリスト。`zone`も`zones`も指定されていない場合、ボリュームは通常、Kubernetesクラスターにノードがあるすべてのアクティブなゾーンにわたってラウンドロビン方式で処理されます。`zone`パラメーターと`zones`パラメーターを同時に使用することはできません。
```

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 9]
```diff
@@ -0,0 +1,538 @@
...
+* `fstype`:`ext4`または`xfs`。デフォルト:`ext4`。定義されたファイルシステムタイプは、ホストオペレーティングシステムでサポートされている必要があります。
+* `replication-type`:`none`または`regional-pd`。デフォルト:`none`。
+
+`replication-type`が`none`に設定されている場合、通常の(ゾーン)PDがプロビジョニングされます。
+
+`replication-type`が`regional-pd`に設定されている場合、[Regional Persistent Disk](https://cloud.google.com/compute/docs/disks/#repds)がプロビジョニングされます。`volumeBindingMode: WaitForFirstConsumer`を設定することを強くお勧めします。この場合、このStorageClassを使用するPersistentVolumeClaimを使用するPodを作成すると、Regional Persistent Diskが2つのゾーンでプロビジョニングされます。1つのゾーンは、Podがスケジュールされているゾーンと同じです。もう1つのゾーンは、クラスターで使用可能なゾーンからランダムに選択されます。ディスクゾーンは、`allowedTopologies`を使用してさらに制限できます。
+
+{{< note >}}
+`zone`および`zones`パラメータは廃止され、[allowedTopologies](#allowed-topologies)に置き換えられました。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/storage-classes.md:241]:
```suggestion
`zone`および`zones`パラメーターは廃止され、[allowedTopologies](#allowed-topologies)に置き換えられました。
```

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 10]
```diff
@@ -0,0 +1,538 @@
...
+* `secretNamespace`、`secretName`:Gluster RESTサービスと通信するときに使用するユーザーパスワードを含むSecretインスタンスの識別。これらのパラメータはオプションです。`secretNamespace`と`secretName`の両方が省略された場合、空のパスワードが使用されます。提供されたシークレットには、タイプ`kubernetes.io/glusterfs`が必要です。たとえば、次のように作成されます。
+    ```
+    kubectl create secret generic heketi-secret \
+      --type="kubernetes.io/glusterfs" --from-literal=key='opensesame' \
+      --namespace=default
+    ```
+
+    シークレットの例は[glusterfs-provisioning-secret.yaml](https://github.com/kubernetes/examples/tree/master/staging/persistent-volume-provisioning/glusterfs/glusterfs-secret.yaml)にあります。
+
+* `clusterid`:`630372ccdc720a92c681fb928f27b53f`は、ボリュームのプロビジョニング時にHeketiによって使用されるクラスターのIDです。また、クラスタIDのリストにすることもできます。これはオプションのパラメータです。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/storage-classes.md:277]:
```suggestion
* `clusterid`:`630372ccdc720a92c681fb928f27b53f`は、ボリュームのプロビジョニング時にHeketiによって使用されるクラスターのIDです。また、クラスターIDのリストにすることもできます。これはオプションのパラメーターです。
```

**Review by nasa9084 — COMMENTED** [review]
/assign

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 11]
```diff
@@ -0,0 +1,538 @@
...
+parameters:
+  repl: "1"
+  snap_interval:   "70"
+  priority_io:  "high"
+
+```
+
+* `fs`:配置するファイルシステム:`none/xfs/ext4`(デフォルト:`ext4`)。
+* `block_size`:キロバイト単位のブロックサイズ(デフォルト:`32`)。
+* `repl`:レプリケーション係数`1..3`の形式で提供される同期レプリカの数(デフォルト:`1`)ここでは文字列が期待されます。つまり、`1`ではなく`"1"`です。
```
[the remark this answers is not in the document]
Comment (KobayashiD27) [reply content/ja/docs/concepts/storage/storage-classes.md]: ありがとうございます。一通り更新しました。

**Review by nasa9084 — COMMENTED** [review]
/approve

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 12]
```diff
@@ -0,0 +1,538 @@
...
+- [CSIプロビジョナー](#vsphere-provisioner-csi):`csi.vsphere.vmware.com`
+- [vCPプロビジョナー](#vcp-provisioner):`kubernetes.io/vsphere-volume`
+
+インツリープロビジョナーは[非推奨です](/blog/2019/12/09/kubernetes-1-17-feature-csi-migration-beta/#why-are-we-migrating-in-tree-plugins-to-csi)。CSIプロビジョナーの詳細については、[Kubernetes vSphere CSIドライバー](https://vsphere-csi-driver.sigs.k8s.io/)および[vSphereVolume CSI移行](/ja/docs/concepts/storage/volumes/#vsphere-csi-migration)を参照してください。
+
+#### CSIプロビジョナー {#vsphere-provisioner-csi}
+
+vSphere CSI StorageClassプロビジョナーは、Tanzu Kubernetesクラスターと連携します。例については、[vSphere CSIリポジトリ](https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/example/vanilla-k8s-RWM-filesystem-volumes/example-sc.yaml)を参照してください。
+
+#### vCPプロビジョナー
```
Comment (s-kawamura-w664) [inline content/ja/docs/concepts/storage/storage-classes.md:346]:
```suggestion
#### vCPプロビジョナー {#vcp-provisioner}
```
This is linked at line 338.

File: content/ja/docs/concepts/storage/storage-classes.md [hunk 13]
```diff
@@ -0,0 +1,538 @@
...
+      image: nginx
+      ports:
+        - containerPort: 80
+          name: "http-server"
+      volumeMounts:
+        - mountPath: "/usr/share/nginx/html"
+          name: task-pv-storage
+```
+
+### 許可されたトポロジー
```
Comment (s-kawamura-w664) [inline content/ja/docs/concepts/storage/storage-classes.md:160]:
```suggestion
### 許可されたトポロジー {#allowed-topologies}
```
This is linked at line 213 and line 241.

**Review by s-kawamura-w664 — COMMENTED** [review]
I left some comments.
They are only the link problem.

File: content/ja/docs/concepts/storage/storage-classes.md (hunk 12 above)
Comment (s-kawamura-w664) [reply content/ja/docs/concepts/storage/storage-classes.md]: I have been able to confirm to jump to the link on netlify preview.

File: content/ja/docs/concepts/storage/storage-classes.md (hunk 13 above)
Comment (s-kawamura-w664) [reply content/ja/docs/concepts/storage/storage-classes.md]: I have been able to confirm to jump to the link on netlify preview.

**Pull request merged**